base_business_document_import/models/business_document_import.py

```diff
@@ -0,0 +1,953 @@
+                        "The VAT number of the customer written in the "
+                        "business document (%s) doesn't match the VAT number "
+                        "of the company '%s' (%s) in which you are trying to "
+                        "import this document.") % (
+                            parsed_company_vat,
+                            company.display_name,
+                            company.partner_id.sanitized_vat))
+            else:
+                chatter_msg.append(_(
+                    "Missing VAT number on company '%s'")
+                    % company.display_name)
+
+    def get_xml_files_from_pdf(self, pdf_file):
+        """Returns a dict with key = filename, value = XML file obj"""
+        logger.info('Trying to find an embedded XML file inside PDF')
+        res = {}
+        try:
+            fd = BytesIO(pdf_file)
+            pdf = PyPDF2.PdfFileReader(fd)
+            logger.debug('pdf.trailer=%s', pdf.trailer)
+            pdf_root = pdf.trailer['/Root']
+            logger.debug('pdf_root=%s', pdf_root)
+            # TODO add support for /Kids
+            embeddedfiles = pdf_root['/Names']['/EmbeddedFiles']['/Names']
+            i = 0
+            xmlfiles = {}  # key = filename, value = PDF obj
+            for embeddedfile in embeddedfiles[:-1]:
+                mime_res = mimetypes.guess_type(embeddedfile)
+                if mime_res and mime_res[0] in ['application/xml', 'text/xml']:
+                    xmlfiles[embeddedfile] = embeddedfiles[i + 1]
+                i += 1
+            logger.debug('xmlfiles=%s', xmlfiles)
+            for filename, xml_file_dict_obj in xmlfiles.items():
+                try:
+                    xml_file_dict = xml_file_dict_obj.getObject()
+                    logger.debug('xml_file_dict=%s', xml_file_dict)
+                    xml_string = xml_file_dict['/EF']['/F'].getData()
+                    xml_root = etree.fromstring(xml_string)
+                    logger.debug(
+                        'A valid XML file %s has been found in the PDF file',
+                        filename)
+                    res[filename] = xml_root
+                except Exception as e:
+                    continue
+        except Exception as e:
+            pass
+        logger.info('Valid XML files found in PDF: %s', list(res.keys()))
+        return res
+
+    @api.model
+    def post_create_or_update(self, parsed_dict, record, doc_filename=None):
+        if parsed_dict.get('attachments'):
+            for filename, data_base64 in\
+                    parsed_dict['attachments'].items():
+                self.env['ir.attachment'].create({
+                    'name': filename,
+                    'res_id': record.id,
+                    'res_model': str(record._name),
+                    'datas': data_base64,
+                    'datas_fname': filename,
+                })
+        for msg in parsed_dict['chatter_msg']:
+            record.message_post(msg)
```

Comment:
Here parameter should be passed explicitly: `body=msg`.
Otherwise I get an error.